websites/N/namu.wiki/presence.ts

```diff
@@ -0,0 +1,177 @@
+  "acl": "View ACl",
+  "diff": "View Difference",
+  "history": "View History",
+  "backlink": "View Backlink",
+  "contribution": "View Contribution",
+  "Policy": "View Policy",
+  "Search": "Searching",
+
+  // edit
+  "edit": "Editing",
+  "move": "Move Document",
+  "delete": "Delete Confirm",
+  "new_edit_request": "Create New Edit Request",
+
+  // discuss
+  "discuss": "Dicsussing",
+  "thread": "View Discuss",
+  "edit_request": "Edit Request",
+
+  // recent
+  "RecentDiscuss": "View Recent Discuss",
+  "RecentChanges": "View Recent Changes",
+
+  // not used well
+  "NeededPages": "View Needed Pages",
+  "OrphanedPages": "View Orphaned Pages",
+  "UncategorizedPages": "View Uncategorized Pages",
+  "OldPages": "View Old Pages",
+  "ShortestPages": "View Shortest Pages",
+  "LongestPages": "View Longest Pages",
+  "BlockHistory": "View Block History",
+  "RandomPage": "View Random Page",
+  "Upload": "Upload file",
+  "License": "View License"
+},
+// /member/(action)
+validateMembershipUrl = /\/member\/(.+)/,
+membersMapping: interfaceMapping = {
+  "login": "Login",
+  "recover_password": "Recover Password",
+  "signup": "Sign Up",
+  "mypage": "My Page",
+  "change_email": "Change Email",
+  "change_password": "Change Password",
+  "activate_otp": "Activate OTP"
+},
+// /RecentChanges?logtype=(search)
+changesMapping: interfaceMapping = {
+  "all": "All Changes",
+  "create": "Created Documents",
+  "delete": "Deleted Documents",
+  "move": "Moved Documents",
+  "revert": "Reverted Documents"
+},
+// /RecentDiscuss?logtype=(search)
+discussMapping: interfaceMapping = {
+  "normal_thread": "Normal Thread",
+  "old_thread": "Old Thread",
+  "closed_thread": "Closed Thread",
+  "open_editrequest": "Opened Edit Request",
+  "accepted_editrequest": "Accepted Edit Request",
+  "closed_editrequest": "Closed Edit Request",
+  "old_editrequest": "Old Edit Request"
+},
+// /contribution/(type)/(username)/(contributeType)
+validateContributeUrl = /\/contribution\/(.+)\/(.+)\/(.+)/;
+
+let currentPage = document.location.pathname, currentTime = Date.now();
+presence.on("UpdateData", async () => {
+  const privacy = await presence.getSetting("privacy"),
+        showTimestamp = await presence.getSetting("showTimestamp"),
+        path = document.location.pathname,
+        params = document.location.search,
+        parsedUrl = path.split("/"), // It's a very bad design, but they have a slash document.
+        action = parsedUrl[1], details = boardTypeMapping[action],
+        presenceData: PresenceData = { largeImageKey: "namu" };
+
+  /**
+   * 
+   * Setting Details & State
+   * 
+   */
+  presenceData.details = details === undefined ? "Unknown Action" : details;
+
+  let page;
+  /* View Contribute */
+  if ((page = validateContributeUrl.exec(path))) {
+    if (page[1] === "author") page = `User: ${page[2]}`;
+    else page = "IP User";
+
+  /* View Membership */
+  } else if ((page = validateMembershipUrl.exec(path))) {
```

Comment:
```suggestion
  if ((page = validateContributeUrl.exec(path)))
    if (page[1] === "author") page = `User: ${page[2]}`;
    else page = "IP User";

  /* View Membership */
  else if ((page = validateMembershipUrl.exec(path))) {
```
![image](https://user-images.githubusercontent.com/9348108/119613722-b4a4b200-bdfd-11eb-9d89-d1fd0aaf4831.png)